dm/dm/master/server.go

```diff
@@ -685,20 +683,21 @@ func extractSources(s *Server, req hasWokers) ([]string, error) {
 			}
 		}
 		if len(invalidSource) > 0 {
-			return nil, errors.Errorf("sources %s haven't been added", invalidSource)
+			return nil, false, errors.Errorf("sources %s haven't been added", invalidSource)
 		}
+		specicySource = true
```

Comment:
I think it's also frighten user when he uses `query-status -s` to check task(subtask) status, but the task is gone.

Reply:
add  a workaround msg in
now when user query a source that bound to an offline worker (no free worker), dmctl will return this:
```
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": false,
            "msg": "[code=38029:class=dm-master:scope=internal:level=high], Message: mysql-01 relevant worker-client not found, Workaround: Please use list-member command to see if the some workers are offline.",
            "sourceStatus": {
                "source": "mysql-01",
                "worker": "",
                "result": null,
                "relayStatus": null
            },
            "subTaskStatus": [
            ]
        }
    ]
}
```

Reply:
> but the task is gone.

the task of this source still not listed.

Reply:
I thought it over again, I think the current state is acceptable to users

* When the user uses `query-status` with no arguments, we list all tasks, regardless of whether the worker processing the task is online or not

* When the user uses `query-status -s xxsource`, he doesn't want to list all tasks, he wants to know the status of a source

When the above exception occurs, there are not enough workers to handle the source requested by the user, so it seems acceptable to return an error like this?

If there's anything else I haven't thought of, please feel free to ask!

Reply:
> When the user uses query-status -s xxsource, he doesn't want to list all tasks, he wants to know the status of a source

if that's right, we should not return `subTaskStatus`. So currently we returning it with a empty value is quite misleading.

Reply:
fill subtaskStatus in https://github.com/pingcap/ticdc/pull/3358/commits/c715b79bcf280228560b60f09bfd4a82c9a9ae98 and https://github.com/pingcap/ticdc/pull/3358/commits/7e2889728ffaf39d52deb59833b70f0eeda02943 when specified source field, ptal😬